background_scripts/completion.js

```diff
@@ -693,7 +693,7 @@ const RankingUtils = {
       let matchedTerm = false;
       for (const thing of things) {
         if (!matchedTerm) {
-          matchedTerm = thing.match(regexp);
+          matchedTerm = thing?.match(regexp);
```

Comment:
This null-check seems redundant if we're populating the url and title with empty strings above, right?

Of the two checks you have, I think I would prefer to allow these fields to be null, document why they can be null, and update the matching code to handle null fields, as you've done here.